entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/type/LocalDateBasicUserType.java

```diff
@@ -18,22 +18,39 @@
 
 import com.blazebit.persistence.view.spi.type.BasicUserType;
 
-import java.sql.Timestamp;
 import java.time.LocalDate;
-import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
 
 /**
- *
  * @author Christian Beikov
  * @since 1.5.0
  */
 public class LocalDateBasicUserType extends TimestampishImmutableBasicUserType<LocalDate> {
 
     public static final BasicUserType<LocalDate> INSTANCE = new LocalDateBasicUserType();
+    private static final DateTimeFormatter OPTIONAL_TIME_FORMATTER = new DateTimeFormatterBuilder()
+            .parseCaseInsensitive()
+            .append(DateTimeFormatter.ISO_LOCAL_DATE)
+            .optionalStart()
+            .appendLiteral(' ')
+            .append(DateTimeFormatter.ISO_LOCAL_TIME)
+            .optionalEnd()
+            .toFormatter();
 
     @Override
     public LocalDate fromString(CharSequence sequence) {
-        return Timestamp.valueOf(sequence.toString()).toInstant().atZone(ZoneOffset.UTC).toLocalDate();
+        String input = sequence.toString();
+        try {
+            return LocalDate.parse(input, OPTIONAL_TIME_FORMATTER);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid date or date-time format");
+        }
     }
 
+    @Override
+    public String toStringExpression(String expression) {
+        return "TO_CHAR(" + expression + ", 'YYYY-MM-DD')";
```

Comment:
I think you will have to revert this change and make the `DateTimeFormatter` smarter if necessary, since the `TO_CHAR` function is not available on all databases.

Reply:
Sounds good, I'll remove the override and will give the `DateTimeFormatter` another look. I had a version of it that supported a lot of different formats so I'll maybe add that back (Handled OffsetDateTime, ZonedDateTime, etc)

Reply:
@beikov Made the suggested changes, however, on the `toStringExpression` comment.

```
@Override
    public String toStringExpression(String expression) {
        return "TO_CHAR(" + expression + ", 'YYYY-MM-DD HH24:MI:SS.US')";
    }
 ```

This is what is in the base class `TimestampishImmutableBasicUserType` so I overrode it and removed the time portion. If we don't want to use TO_CHAR, this should probably be changed and we should provide one without the time, but I'm not clear on what/how this is used so maybe it's better if that's something someone with more knowledge could address unless you can provide some guidance there.